compiling errors out on mac os

With the changes in: https://github.com/opendnssec/SoftHSMv2/commit/f5dcf2d4646785b0b37a0e75eab146cf39360cc3
I am getting the following errors and warning:
./PublicKey.h:64:23: warning: 'constexpr' non-static member function will not be
      implicitly 'const' in C++14; add 'const' to avoid a change in behavior
      [-Wconstexpr-not-const]
        constexpr PublicKey& operator=(const PublicKey&) { return *this; }
                             ^
                                                         const
./PublicKey.h:64:60: error: binding value of type 'const PublicKey' to reference
      to type 'PublicKey' drops 'const' qualifier
        constexpr PublicKey& operator=(const PublicKey&) { return *this; }
                                                                  ^~~~~
./PublicKey.h:64:23: error: no return statement in constexpr function
        constexpr PublicKey& operator=(const PublicKey&) { return *this; }
                             ^
In file included from AESKey.cpp:37:
In file included from ./CryptoFactory.h:40:
In file included from ./AsymmetricAlgorithm.h:37:
In file included from ./AsymmetricKeyPair.h:39:
./PrivateKey.h:71:24: warning: 'constexpr' non-static member function will not
      be implicitly 'const' in C++14; add 'const' to avoid a change in behavior
      [-Wconstexpr-not-const]
        constexpr PrivateKey& operator=(const PrivateKey&) { return *this; }
                              ^
                                                           const
./PrivateKey.h:71:62: error: binding value of type 'const PrivateKey' to
      reference to type 'PrivateKey' drops 'const' qualifier
        constexpr PrivateKey& operator=(const PrivateKey&) { return *this; }
                                                                    ^~~~~
./PrivateKey.h:71:24: error: no return statement in constexpr function
        constexpr PrivateKey& operator=(const PrivateKey&) { return *this; }
                              ^
2 warnings and 4 errors generated.
make[5]: *** [AESKey.lo] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

--- a/src/lib/crypto/PrivateKey.h
+++ b/src/lib/crypto/PrivateKey.h
@@ -68,7 +68,7 @@ public:
 	virtual bool PKCS8Decode(const ByteString& ber) = 0;
 
 	// Assignment
-	constexpr PrivateKey& operator=(const PrivateKey&) { return *this; }
+	constexpr const PrivateKey& operator=(const PrivateKey&) const { return *this; }
 };
 
 #endif // !_SOFTHSM_V2_PRIVATEKEY_H

--- a/src/lib/crypto/PublicKey.h
+++ b/src/lib/crypto/PublicKey.h
@@ -61,7 +61,7 @@ public:
 	virtual ByteString serialise() const = 0;
 
 	// Assignment
-	constexpr PublicKey& operator=(const PublicKey&) { return *this; }
+	constexpr const PublicKey& operator=(const PublicKey&) const { return *this; }
 };
 
 #endif // !_SOFTHSM_V2_PUBLICKEY_H

